### PR TITLE
TTdilepDraw clean up

### PR DESCRIPTION
**Pull request by tt8888tt**
added drawing code for step6 plots(top, ttbar etc)

### CODE REVIEW

File: CatAnalyzer/test/TTdilepDraw/TTdilepDraw.py [hunk 1]
```diff
@@ -1,21 +1,32 @@
 import ROOT,copy
 import array
-import sys
 
 def hist_maker(name, title, bin_set, x_name, y_name, tr, br, cut):
-    if title == "ll_m":
-        massbin = array.array('d', [20, 30, 45, 75, 105, 125, 165, 260, 400])
-        hist = ROOT.TH1F(name, title, 8, massbin)
-    else: hist = ROOT.TH1F(name, title, bin_set[0], bin_set[1], bin_set[2])
+    hist = ROOT.TH1F(name, title, bin_set[0], bin_set[1])
```
Comment (jhgoh) [inline CatAnalyzer/test/TTdilepDraw/TTdilepDraw.py]:
``` python
hist = ROOT.TH1F(name, title. len(bins)-1, bins)
```

or if you pass list,

``` python
hist = ROOT.TH1F(name, title, len(bins)-1, array('f', bins))
```

File: CatAnalyzer/test/TTdilepDraw/TTdilepDraw.py [hunk 2]
```diff
@@ -23,118 +34,141 @@ def hist_maker(name, title, bin_set, x_name, y_name, tr, br, cut):
...
+
+massbin = [8, array.array('d', [20, 30, 45, 75, 105, 125, 165, 260, 400])]
+njetbin = [10, array.array('d', [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10])]
+metbin = [10, array.array('d', [0, 30, 60, 90, 120, 150, 180, 210, 240, 270, 300])]
+nbjetbin = [6, array.array('d', [0, 1, 2, 3, 4, 5, 6])]
+topptbin = [4, array.array('d', [0, 70, 140, 240, 400])]
+toprapibin = [5, array.array('f', [0.0, 0.25, 0.65, 1.15, 1.7, 2.5])]
+ptbin = [5, array.array('d', [20, 40, 70, 120, 180, 400])]
+etabin = [8, array.array('f', [-2.5, -1.5, -1, -0.5, 0, 0.5, 1, 1.5, 2.5])]
+jetptbin = [5, array.array('d', [30, 50, 80, 130, 210, 500])]
```
Comment (jhgoh) [inline CatAnalyzer/test/TTdilepDraw/TTdilepDraw.py]: You don't need bin size

**Comment by jhgoh** [comment]
In general, you can use HistMaker.py to "project" your ntuples, which is generic pyROOT script which covers to do normalization by cross section, etc.

**Comment by tt8888tt** [comment]
I see. I will look it up after kps. I modified some about binset just for now.

File: CatAnalyzer/test/TTdilepDraw/TTdilepDraw.py [hunk 3]
```diff
@@ -1,21 +1,32 @@
...
+
+def hist_maker(name, title, bins, x_name, y_name, tr, br, cut):
+    hist = ROOT.TH1F(name, title, len(bins)-1, array('f', bins))
     hist.SetStats(0)
     tr.Project(name, br, cut)
+    if "1" in title:
+        hist2 = ROOT.TH1F(name+"2", title, len(bins)-1, array('f', bins))
+        br = plotvar.replace("1", "2")
+        tr.Project(name+"2", br, cut)
+        hist.Add(hist2)
```
Comment (jhgoh) [inline CatAnalyzer/test/TTdilepDraw/TTdilepDraw.py:14]: Is it safe? What happens if title contains "1" with different meaning - say hLepton1Pt etc.. Wouldn't it better to check suffix only?

File: CatAnalyzer/test/TTdilepDraw/TTdilepDraw.py [hunk 4]
```diff
@@ -23,118 +34,140 @@ def hist_maker(name, title, bin_set, x_name, y_name, tr, br, cut):
               'WW_TuneCUETP8M1_13TeV',
               'WZ_TuneCUETP8M1_13TeV',
               'DYJetsToLL_M-50']
-mcval_l = [831.8, 61526.7, 35.6, 35.6, 31.8, 65.9, 118.7, 6025.2]
-color_l = [2, 0, 3, 5, 7, 7, 7, 4]
-
 rdfilelist = ['MuonEG']
 
+mcval_l = [815.96, 61526.7, 35.6, 35.6, 15.4, 110.8, 66.1, 6025.2]
+nevent_l = [19899500, 24151270, 1000000, 995600, 996168, 994416, 991232, 28825132]
```
Comment (jhgoh) [inline CatAnalyzer/test/TTdilepDraw/TTdilepDraw.py]: These info can be retrieved from the samples.json file. do you want me to modify these lines to use it?

Comment (jshlee) [inline CatAnalyzer/test/TTdilepDraw/TTdilepDraw.py]: i dont think DJ knows how to do that yet

File: CatAnalyzer/test/TTdilepDraw/TTdilepDraw.py (hunk 3 above)
Comment (tt8888tt) [inline CatAnalyzer/test/TTdilepDraw/TTdilepDraw.py:14]: Sorry. I don't follow... This is to add 'lep2_pt' histogram to 'lep1_pt'. are you suggesting that I should change name of value in tree..?

File: CatAnalyzer/test/TTdilepDraw/TTdilepDraw.py (hunk 4 above)
Comment (tt8888tt) [inline CatAnalyzer/test/TTdilepDraw/TTdilepDraw.py]: sure I didn't know, but I think I kinda figure out... not sure though.

**Comment by tt8888tt** [comment]
but clearly, google doc should be fully filled to use this... v7-4-3 doesn't seem like.

**Pull request closed without merging**